cli/build.go

```diff
@@ -726,8 +716,3 @@ func DownloadYB() (string, error) {
 
 	return ybPath, nil
 }
-
-func DeleteLocalYB() error {
```

Comment:
I don't agree with this. Getting yb from the correct channel and auto-detecting devmode don't invalidate the need to cleanup this temporary directory. Maybe we could use a real temporary directory by using `os.TempDir()` (https://godoc.org/os#TempDir), as most modern O.S. clean them up at boot or shutdown.

Reply:
Seems reasonable -- I changed it to use our cache dir, also checked to see if the binary checksum matches the known good one before bothering to try re-downloading